templates/nextflow-aurora-mysql8.yaml

```diff
@@ -0,0 +1,461 @@
+    Properties:
+      ServiceToken: !GetAtt BootstrapLambdaFunction.Arn
+
+  BootstrapLambdaRole:
+    Condition: UseSnapshotFalse
+    Type: AWS::IAM::Role
+    Properties:
+      AssumeRolePolicyDocument:
+        Version: "2012-10-17"
+        Statement:
+          - Effect: Allow
+            Principal:
+              Service:
+              - lambda.amazonaws.com
+            Action:
+              - sts:AssumeRole
+      ManagedPolicyArns:
+        - 'arn:aws:iam::aws:policy/service-role/AWSLambdaVPCAccessExecutionRole'
+
+  BootstrapLambdaPolicy:
+    Condition: UseSnapshotFalse
+    Type: AWS::IAM::ManagedPolicy
+    Properties:
+      Description: Allow access to SecretsManager and RDS for Bootstrap Lambda Role
+      PolicyDocument:
+        Version: 2012-10-17
+        Statement:
+        - Effect: Allow
+          Action: secretsmanager:GetSecretValue
+          Resource:
+            - !Ref AuroraMasterSecret
+            - !Ref AuroraNextflowTowerDatabaseUserSecret
+        - Effect: Allow
+          Action: 'rds-data:*'
+          Resource: '*'
+      Roles:
+        - !Ref BootstrapLambdaRole
+
+  BootstrapLambdaFunction:
+    Condition: UseSnapshotFalse
+    Type: AWS::Lambda::Function
+    Properties:
+      Description: Bootstrap newly Created Aurora Database
+      Handler: index.lambda_handler
+      Role: !GetAtt BootstrapLambdaRole.Arn
+      Runtime: python3.10
+      Environment:
+        Variables:
+          CLUSTER_ARN: !Sub 'arn:aws:rds:${AWS::Region}:${AWS::AccountId}:cluster:${AuroraCluster}'
+          MASTER_SECRET_ARN: !Ref AuroraMasterSecret
+          DB_NAME: !Ref DBName
+          DB_USER_SECRET_ARN: !Ref AuroraNextflowTowerDatabaseUserSecret
+      Code:
+        ZipFile: |
+          import boto3
+          import cfnresponse
+          import json
+          import logging
+          from os import environ
+
+          log = logging.getLogger(__name__)
+          log.setLevel(logging.DEBUG)
+
+          def submit_sql(sql):
+            return boto3.client('rds-data').execute_statement(
+              secretArn=environ['MASTER_SECRET_ARN'],
+              database=environ['DB_NAME'],
+              resourceArn=environ['CLUSTER_ARN'],
+              sql=sql
+            )
+
+
+          def bootstrap():
+            try:
+              db_name = environ['DB_NAME']
+              secret_response = boto3.client('secretsmanager').get_secret_value(SecretId=environ['DB_USER_SECRET_ARN'])
+              secret = json.loads(secret_response['SecretString'])
+              username = secret['username']
+              password = secret['password']
+              alter_db_response = submit_sql(f"ALTER DATABASE {db_name} CHARACTER SET utf8 COLLATE utf8_bin;")
+              log.debug(f"Alter DB response: {alter_db_response}")
+              create_user_response = submit_sql(f"CREATE USER IF NOT EXISTS '{username}' IDENTIFIED BY '{password}';")
+              log.debug(f"Create user response: {create_user_response}")
```

Comment:
Is it possible that this could leak the password if an exception was raised during the `return boto3.client('rds-data').execute_statement` call?

Reply:
No, the error that is returned would be something like this.. `An error was encountered when bootstrapping: An error occurred (DatabaseErrorException) when calling the ExecuteStatement operation: Operation CREATE USER failed for 'tower'@'%'; Error code: 1396; SQLState: HY000`